### PR TITLE
Shipyard RCD

### PR DESCRIPTION
**Pull request by GreaseMonk**
## About the PR

This adds a RCD that only works on a ship for which it is authorized.
the Shipyard RCD (SRCD) works as follows:

- Can be purchased at EngiVend for 5k$ and 1k$ for its ammo
- By default it is useless when unauthorized.
- Needs to be authorized by swiping the captain's id card on it.
- Can then optionally be shared with crewmembers
- Has it's own special ammo
- Normal ammo does NOT work on it.
- Can be researched at tier 2 for 10k industrial
- Can be made in the protolathe, as well as it's ammo , following this reseach.

## Why / Balance

This adds a RCD that can't be used on POI's, while also enabling ship and station engineers to make quick modifications to their own ship, or, build new stations quickly, as ship and station building is a part of frontier station!

## Technical details

Made yml copies of rcd and rcd ammo and flagged them up, and modified rcdsystem so it works based on these flags.
Also added the research for them.
Needed to make deed networked

## Media
https://github.com/new-frontiers-14/frontier-station-14/assets/1354802/5524c4a8-f7bb-40fb-9813-2eee11e85392

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
none

**Changelog**
add: Shipyard RCD

### CODE REVIEW

**Comment by Cheackraze** [comment]
error from the mapchecker *should* be fixed if you merge master now.

File: Content.Shared/RCD/Systems/RCDAmmoSystem.cs [hunk 1]
```diff
@@ -42,6 +42,16 @@ private void OnAfterInteract(EntityUid uid, RCDAmmoComponent comp, AfterInteract
             return;
 
         var user = args.User;
+
+        // ## Frontier - Shipyard RCD ammo only fits in shipyard RCD.
+        // At this point RCDComponent is guaranteed
+        EnsureComp<RCDComponent>(target, out var rcdComponent);
```
Comment (Cheackraze) [inline Content.Shared/RCD/Systems/RCDAmmoSystem.cs:48]: optionally you could change the HasComp above to a try comp and out the rcdComponent, but for the sake of code seperation this is fine, and its not called often anyway to worry about query calls.

File: Content.Shared/RCD/Systems/RCDSystem.cs [hunk 2]
```diff
@@ -162,14 +253,6 @@ private void OnDoAfter(EntityUid uid, RCDComponent comp, RCDDoAfterEvent args)
...
         var snapPos = mapGrid.TileIndicesFor(location);
 
-        // I love that this uses entirely separate code to construction and tile placement!!!
-
-        var gridUid = mapGrid.Owner;
-        var ev = new FloorTileAttemptEvent();
-
-        if (HasComp<ProtectedGridComponent>(gridUid) || ev.Cancelled) // Frontier - Remove all RCD use on outpost.
-            return;
-
```
Comment (Cheackraze) [inline Content.Shared/RCD/Systems/RCDSystem.cs:172]: if we leave normal RCDs as obtainable, this should remain

**Review by Cheackraze — COMMENTED** [review]
the only actual thing that needs changing is leaving the protected component on the station just in case

**Review by dvir001 — APPROVED** [review]
Its working as intended.

**Pull request merged**